x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java

```diff
@@ -0,0 +1,288 @@
+        assertThat(updatedTimestampMillisRange.isComplete(), equalTo(true));
+        assertThat(updatedTimestampMillisRange, not(sameInstance(IndexLongFieldRange.EMPTY)));
+        assertThat(updatedTimestampMillisRange.getMin(), greaterThanOrEqualTo(Instant.parse("2020-11-26T00:00:00Z").getMillis()));
+        assertThat(updatedTimestampMillisRange.getMax(), lessThanOrEqualTo(Instant.parse("2020-11-27T00:00:00Z").getMillis()));
+        assertThat(indicesService.getTimestampFieldType(updatedIndexMetadata.getIndex()), notNullValue());
+
+        // Stop the node holding the searchable snapshots, and since we defined
+        // the index allocation criteria to require the searchable snapshot
+        // index to be allocated in that node, the shards should remain unassigned
+        internalCluster().stopNode(dataNodeHoldingSearchableSnapshot);
+        waitUntilAllShardsAreUnassigned(updatedIndexMetadata.getIndex());
+
+        if (includeIndexCoveringSearchRangeInSearchRequest) {
+            SearchResponse newSearchResponse = client().search(request).actionGet();
+
+            assertThat(newSearchResponse.getSkippedShards(), equalTo(indexOutsideSearchRangeShardCount));
+            assertThat(newSearchResponse.getSuccessfulShards(), equalTo(totalShards));
+            assertThat(newSearchResponse.getFailedShards(), equalTo(0));
+            assertThat(newSearchResponse.getTotalShards(), equalTo(totalShards));
+        } else {
+            if (indexOutsideSearchRangeShardCount == 1) {
+                expectThrows(SearchPhaseExecutionException.class, () -> client().search(request).actionGet());
+            } else {
+                SearchResponse newSearchResponse = client().search(request).actionGet();
+                // When all shards are skipped, at least one of them should be queried in order to
+                // provide a proper search response.
+                assertThat(newSearchResponse.getSkippedShards(), equalTo(indexOutsideSearchRangeShardCount - 1));
+                assertThat(newSearchResponse.getSuccessfulShards(), equalTo(indexOutsideSearchRangeShardCount - 1));
+                assertThat(newSearchResponse.getFailedShards(), equalTo(1));
+                assertThat(newSearchResponse.getTotalShards(), equalTo(indexOutsideSearchRangeShardCount));
+            }
+        }
+    }
+
+    private void createIndexWithTimestamp(String indexName, int numShards, Settings extraSettings) throws IOException {
+        assertAcked(
+            client().admin()
+                .indices()
+                .prepareCreate(indexName)
+                .setMapping(
+                    XContentFactory.jsonBuilder()
+                        .startObject()
+                        .startObject("properties")
+                        .startObject(DataStream.TimestampField.FIXED_TIMESTAMP_FIELD)
+                        .field("type", "date_nanos")
```

Comment:
Can we randomly choose between `date` and `date-nanos`?

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/65583/commits/eada99ea3b08b21ef9ce5a39e93a305331bf9117